api/encoding/inbound_call.go

```diff
@@ -42,7 +43,34 @@ type inboundCallKey struct{} // context key for *InboundCall
 //
 // A request context is returned and must be used in place of the original.
 func NewInboundCall(ctx context.Context) (context.Context, *InboundCall) {
+	return NewInboundCallWithOptions(ctx)
+}
+
+// InboundCallOption is an option for configuring an InboundCall
+type InboundCallOption interface {
+	apply(call *InboundCall)
+}
+
+type inboundCallOptionFunc func(*InboundCall)
+
+func (i inboundCallOptionFunc) apply(call *InboundCall) { i(call) }
+
+// DisableResponseHeaders disables response headers for inbound calls.
+func DisableResponseHeaders() InboundCallOption {
```

Comment:
No action needed right now: We should probably pass this for oneway too where
relevant since you can't write response headers for oneway requests either.